### ISSUE TITLE
Compatible with PRO and English version

### ISSUE THREAD

**Issue by YorhaX2P** [underecho/DiscordGameSDK4CSP#4]
Would this work with the PRO and English version? How I do install this, I can't find any plugin in the CSP location folder.

**Comment by Mingggggggggggggg** [comment]
There are no plugins for global Clip Studio Paint Users. However the Japanese CSP Ex Version supports plugins. And as far as I know this solution here doesn't work anymore

**Comment by underecho** [comment]
Back in the days of CSP v1, the Pro version worked with a little trickery, but I haven't checked what happened in v2.
The next update will automatically search memory, so it will probably work, but it may take a little longer due to the work I have taken on.

**Comment by YorhaX2P** [comment]
> Back in the days of CSP v1, the Pro version worked with a little trickery, but I haven't checked what happened in v2. The next update will automatically search memory, so it will probably work, but it may take a little longer due to the work I have taken on.

I am still using v1.
I found there is one PlugIn folder, Would place DLL in CLIP STUDIO PAINT or just the PlugIn folder? Because README makes me confused a bit.
![image](https://github.com/underecho/DiscordGameSDK4CSP/assets/16865563/4a59394b-07c0-4818-b259-084fd82865b7)

**Comment by underecho** [comment]
As for the dll files, please put them in the "Plugin/PAINT" folder.

However, some versions of CSP are quite unstable and have not been tested with the Global version, so it would be helpful if you could tell me the version if it doesn't work.

(sorry im using deepl)

**Comment by YorhaX2P** [comment]
> As for the dll files, please put them in the "Plugin/PAINT" folder.
> 
> However, some versions of CSP are quite unstable and have not been tested with the Global version, so it would be helpful if you could tell me the version if it doesn't work.
> 
> (sorry im using deepl)

I put discordRPC4CSP.dll to Plugin/PAINT/discordRPC4CSP.dll folder

And happened this error,
![image](https://github.com/underecho/DiscordGameSDK4CSP/assets/16865563/10bdc29a-f416-4257-bde6-813413b8fd19)

**Comment by Mingggggggggggggg** [comment]
Does it mean it should work on 2.0 or 2.X? I could try it tomorrow but I only have 2.0 Ex

**Comment by underecho** [comment]
TL;DR
**Currently it is completely broken. I'll fix it, but I need to wait a month cuz I have job to do.**

CSP v1
- work on some old version
- not tested global version
- maybe broken on newer version (thx @ SametChan)

CSP v2
- 99% not working
- I don't have v2

Todo
- I'll buy CSP v2
- Try memory search algorithms

**Comment by WildRiley** [comment]
Is this still being worked on? I am getting the same issue on CSP 4.0.3 EX.
The error being:
```
The latest version of PAINT is required to use plugins with the following file names.
Update PAINT to the latest version to use these.

'discordRPC4CSP.dll"
```

**Comment by underecho** [comment]
@WildRiley and ALL
I finally bought the new version of CSP recently.

However, since it’s the Pro version, I’m not sure if it supports Ex.
I’ll give it a try, so please bear with me for a moment.

**Comment by RileyIsPurple** [comment]
@YorhaX2P 
Why is this issue now closed? Is it fixed?

**Comment by underecho** [comment]
@RileyIsPurple 
hi, I think it's probably because my response was slow and it was no longer necessary for him.
I'll create a separate issue, so please wait a moment.

**Comment by underecho** [comment]
so, I found simular project
https://github.com/kakuun333/CspDiscordRpcGd
If you need it right away, we recommend using this project.

Also, to be honest, this is just an experimental project I created to study DLL injection, so I have no motivation to fix it.
Since there are alternatives available, I’ve lost even more motivation, so it probably won’t be updated. I’m sorry.
